tests/queries/0_stateless/00976_system_stop_ttl_merges.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+# shellcheck source=./merges.lib
+. "$CURDIR"/merges.lib
+set -e
+
+$CLICKHOUSE_CLIENT --query "drop table if exists ttl;"
+
+$CLICKHOUSE_CLIENT --query "create table ttl (d Date, a Int) engine = MergeTree order by a partition by toDayOfMonth(d) ttl d + interval 1 day;"
+
+$CLICKHOUSE_CLIENT --query "system stop ttl merges ttl;"
+
+$CLICKHOUSE_CLIENT --query "insert into ttl values (toDateTime('2000-10-10 00:00:00'), 1), (toDateTime('2000-10-10 00:00:00'), 2);"
+$CLICKHOUSE_CLIENT --query "insert into ttl values (toDateTime('2100-10-10 00:00:00'), 3), (toDateTime('2100-10-10 00:00:00'), 4);"
+
+$CLICKHOUSE_CLIENT --query "optimize table ttl partition 10 final;"
+$CLICKHOUSE_CLIENT --query "select * from ttl order by d, a;"
+
+$CLICKHOUSE_CLIENT --query "system start ttl merges ttl;"
+$CLICKHOUSE_CLIENT --query "optimize table ttl partition 10 final;"
+
+wait_for_merges_done ttl
```

Comment:
OPTIMIZE query is synchronous by default, we don't need to wait for merges

Reply:
> OPTIMIZE query is synchronous by default, we don't need to wait for merges

Here is the explanation what is happening.
https://github.com/ClickHouse/ClickHouse/issues/44982
In a short version
- optimize final failes due to concurrent background merge
- next request is concurrent with background merge and sees data before that merge

Reply:
But there are no concurrent background merges in 00976_system_stop_ttl_merges (afaik)

Reply:
Also it's not enough to check `system.merges` as `wait_for_merges_done` does anyway.